src/ripple/app/misc/ValidatorList.h

```diff
@@ -20,102 +20,424 @@
 #ifndef RIPPLE_APP_MISC_VALIDATORLIST_H_INCLUDED
 #define RIPPLE_APP_MISC_VALIDATORLIST_H_INCLUDED
 
-#include <ripple/basics/BasicConfig.h>
+#include <ripple/app/misc/detail/Work.h>
+#include <ripple/app/misc/Manifest.h>
 #include <ripple/basics/Log.h>
+#include <ripple/basics/StringUtilities.h>
 #include <ripple/basics/UnorderedContainers.h>
+#include <ripple/crypto/csprng.h>
 #include <ripple/protocol/PublicKey.h>
-#include <boost/optional.hpp>
-#include <functional>
-#include <memory>
+#include <boost/asio.hpp>
+#include <boost/iterator/counting_iterator.hpp>
 #include <mutex>
 
 namespace ripple {
 
+/**
+    Trusted Validators List
+    -----------------------
+
+    Rippled accepts ledger proposals and validations from trusted validator
+    nodes. A ledger is considered fully-validated once the number of received
+    trusted validations for a ledger meets or exceeds a quorum value.
+
+    This class manages the set of validation public keys the local rippled node
+    trusts. The list of trusted keys is populated using the keys listed in the
+    configuration file as well as remote sites that serve lists of keys
+    recommended by trusted publishers. The site URIs and trusted publisher
+    public keys are specified in the config.
+
+    The validator lists from the remote sites are fetched at a regular interval.
+    Fetched lists are expected to be in JSON format and contain the following
+    fields:
+
+    @li @c "manifest": Base64-encoded serialization of a manifest containing the
+        publisher's master and signing public keys. This has the same format as
+        the @c validation_manifest config section.
+
+    @li @c "blob": Base64-encoded JSON string containing a @c "sequence" and
+        @c "validators" field. @c "validators" contains an array of objects with
+        @c "validation_public_key" and @c "manifest" fields.
+        @c "validation_public_key" must be an Ed25519 master public key.
```

Comment:
Is there any particular reason we require an Ed25519 public key? It would be much nicer if we did everything with a generic key/signature type. That way if we want to add another algorithm, we only need to add it in one place rather than everywhere we have public keys and signatures. I thought we already went through the effort of having generic handlers for keys/signatures.